terraform/charm/sharded/variables.tf

```
@@ -0,0 +1,82 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# variable "cluster_name" {
+#   type        = string
+#   description = "The cluster name of the fleet."
+#   default     = "opensearch"
+# }
+
+variable "config-server" {
+  description = "Config server orchestrator app definition"
+  type = object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), {role="shard"})
+    channel           = optional(string, "6/stable")
+    base              = optional(string, "ubuntu@22.04")
+    revision          = optional(string, 199)
+    units             = optional(number, 3)
+    constraints       = optional(string, "arch=amd64")
+    machines          = optional(list(string), [])
+    storage           = optional(map(string), {})
+    endpoint_bindings = optional(map(string), {})
+    expose            = optional(bool, false)
+  })
+}
+
+variable "sharded" {
+  description = "Shard orchestrator app definition"
+  type = object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), {role="shard"})
+    channel           = optional(string, "6/stable")
+    base              = optional(string, "ubuntu@22.04")
+    revision          = optional(string, 199)
```

Use the default value coming from the simple_deployment module.
```suggestion
    revision          = optional(number, null)
```